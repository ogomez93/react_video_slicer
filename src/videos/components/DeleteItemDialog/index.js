import React, { Fragment } from 'react';
import Button from '@material-ui/core/Button';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

const DeleteItemDialog = ({
  dialogDescription,
  dialogTitle,
  end,
  name,
  start,
  onClose,
  deleteItem
}) => (
  <Fragment>
    <DialogTitle id={dialogTitle}>
      Are you sure you want to delete this clip?
    </DialogTitle>
    <DialogContent>
      <DialogContentText id={dialogDescription}>
        {`Name: '${name}', from ${start} to ${end}`}
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onClose} color="primary">
        Cancel
      </Button>
      <Button onClick={deleteItem} color="secondary">
        Delete
      </Button>
    </DialogActions>
  </Fragment>
);

export default DeleteItemDialog
