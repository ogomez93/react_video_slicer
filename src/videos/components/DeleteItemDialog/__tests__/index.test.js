import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import DeleteItemDialog from '../index';

const dialogDescription = 'dialogDescription';
const dialogTitle = 'dialogTitle';
const end = '00:48';
const name = 'name';
const start = '00:30';
const onClose = jest.fn();
const deleteItem = jest.fn();
const props = {
  dialogDescription,
  dialogTitle,
  end,
  name,
  start,
  onClose,
  deleteItem
}

describe('Videos / Components / DeleteItemDialog', () => {
  let wrapper, shallow;

  beforeAll(() => {
    shallow = createShallow();
    wrapper = shallow(<DeleteItemDialog {...props} />);
    console.log(wrapper.debug());
  });

  it('should render', () =>
    expect(wrapper.isEmptyRender()).toEqual(false));

  it('should match the snapshot', () =>
    expect(wrapper).toMatchSnapshot());
});
