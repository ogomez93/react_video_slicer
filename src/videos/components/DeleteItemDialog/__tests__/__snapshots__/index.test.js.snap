// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos / Components / DeleteItemDialog should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteItemDialog
    deleteItem={[MockFunction]}
    dialogDescription="dialogDescription"
    dialogTitle="dialogTitle"
    end="00:48"
    name="name"
    onClose={[MockFunction]}
    start="00:30"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(DialogTitle)
          id="dialogTitle"
        >
          Are you sure you want to delete this clip?
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)
            id="dialogDescription"
          >
            Name: 'name', from 00:30 to 00:48
          </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="primary"
            id="deleteItemDialogCancel"
            onClick={[MockFunction]}
          >
            Cancel
          </WithStyles(Button)>
          <WithStyles(Button)
            color="secondary"
            id="deleteItemDialogDelete"
            onClick={[MockFunction]}
          >
            Delete
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Are you sure you want to delete this clip?",
          "id": "dialogTitle",
        },
        "ref": null,
        "rendered": "Are you sure you want to delete this clip?",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(DialogContentText)
            id="dialogDescription"
          >
            Name: 'name', from 00:30 to 00:48
          </WithStyles(DialogContentText)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Name: 'name', from 00:30 to 00:48",
            "id": "dialogDescription",
          },
          "ref": null,
          "rendered": "Name: 'name', from 00:30 to 00:48",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              id="deleteItemDialogCancel"
              onClick={[MockFunction]}
            >
              Cancel
            </WithStyles(Button)>,
            <WithStyles(Button)
              color="secondary"
              id="deleteItemDialogDelete"
              onClick={[MockFunction]}
            >
              Delete
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel",
              "color": "primary",
              "id": "deleteItemDialogCancel",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Delete",
              "color": "secondary",
              "id": "deleteItemDialogDelete",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Delete",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(DialogTitle)
            id="dialogTitle"
          >
            Are you sure you want to delete this clip?
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
              id="dialogDescription"
            >
              Name: 'name', from 00:30 to 00:48
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              id="deleteItemDialogCancel"
              onClick={[MockFunction]}
            >
              Cancel
            </WithStyles(Button)>
            <WithStyles(Button)
              color="secondary"
              id="deleteItemDialogDelete"
              onClick={[MockFunction]}
            >
              Delete
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Are you sure you want to delete this clip?",
            "id": "dialogTitle",
          },
          "ref": null,
          "rendered": "Are you sure you want to delete this clip?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)
              id="dialogDescription"
            >
              Name: 'name', from 00:30 to 00:48
            </WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Name: 'name', from 00:30 to 00:48",
              "id": "dialogDescription",
            },
            "ref": null,
            "rendered": "Name: 'name', from 00:30 to 00:48",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                id="deleteItemDialogCancel"
                onClick={[MockFunction]}
              >
                Cancel
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="secondary"
                id="deleteItemDialogDelete"
                onClick={[MockFunction]}
              >
                Delete
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel",
                "color": "primary",
                "id": "deleteItemDialogCancel",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Delete",
                "color": "secondary",
                "id": "deleteItemDialogDelete",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Delete",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
